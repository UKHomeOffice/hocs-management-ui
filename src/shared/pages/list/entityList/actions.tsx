import EntityListItem from 'shared/models/entityListItem';

export type SetEntityDetails = {
    type: 'SetItemDetails';
    payload: EntityListItem;
};

export type SetSimpleName = {
    type: 'SetSimpleName';
    payload: string;
};

export type SetTitle = {
    type: 'SetTitle';
    payload: string;
};

export type PopulateEntities = {
    payload: EntityListItem[];
    type: 'PopulateEntities';
};

export type ToggleShowInactive = {
    payload: boolean;
    type: 'ToggleShowInactive';
};

export type Action =
    SetEntityDetails |
    PopulateEntities |
    SetTitle |
    SetSimpleName |
    ToggleShowInactive;
