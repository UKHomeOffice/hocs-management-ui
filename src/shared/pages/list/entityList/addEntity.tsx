import React, { Reducer } from 'react';
import { RouteComponentProps } from 'react-router';
import { Link } from 'react-router-dom';
import { object, string } from 'yup';
import Submit from '../../../common/components/forms/submit';
import Text from '../../../common/components/forms/text';
import { ApplicationConsumer } from '../../../contexts/application';
import { createListItem } from '../../../services/entityListService';
import { reducer } from './addEntityReducer';
import ErrorSummary from '../../../common/components/errorSummary';
import {
    GENERAL_ERROR_TITLE,
    VALIDATION_ERROR_TITLE
} from '../../../models/constants';
import useError from '../../../hooks/useError';
import ErrorMessage from '../../../models/errorMessage';
import EntityListItem from '../../../models/entityListItem';
import { validate } from '../../../validation';
import InputEventData from '../../../models/inputEventData';

interface AddCampaignProps extends RouteComponentProps {
    csrfToken?: string;
    entityDefinition: EntityDefinition;
}

const AddEntity: React.FC<AddCampaignProps> = ({ csrfToken, history, entityDefinition }) => {

    const [pageError, addFormError, clearErrors, setErrorMessage] = useError('', VALIDATION_ERROR_TITLE);
    const [entity, dispatch] = React.useReducer<Reducer<EntityListItem, InputEventData>>(reducer, {
        uuid: '',
        title: '',
        simpleName: ''
    });

    const validationSchema = object({
        title: string()
            .required()
            .label(`${entityDefinition.entityNameCapitalised} name`)
            .matches(/^[a-zA-Z0-9_,.!? ()&]*$/),
        simpleName: string()
            .required()
            .label(`${entityDefinition.entityNameCapitalised} code`)
            .matches(/^[a-zA-Z0-9_,.!? ()&]*$/)
    });

    const handleSubmit = (event: React.FormEvent) => {
        event.preventDefault();
        clearErrors();
        if (validate(validationSchema, entity, addFormError)) {
            createListItem(entity, entityDefinition.entityListName).then(() => {
                history.push('/', { successMessage: entityDefinition.messages.ADD_ENTITY_SUCCESS });
            }).catch((error) => {
                if (error && error.response && error.response.status === 409) {
                    setErrorMessage(new ErrorMessage(entityDefinition.messages.DUPLICATE_ENTITY_ERROR_DESCRIPTION, VALIDATION_ERROR_TITLE));
                } else {
                    setErrorMessage(new ErrorMessage(entityDefinition.messages.ADD_ENTITY_ERROR_DESCRIPTION, GENERAL_ERROR_TITLE));
                }
            });
        }
    };

    return (
        <>
            <div className="govuk-grid-row">
                <div className="govuk-grid-column-two-thirds-from-desktop">
                    <Link to={entityDefinition.entityRoute} className="govuk-back-link">Back</Link>
                    <ErrorSummary
                        pageError={pageError}
                    />
                    <h1 className="govuk-heading-xl">
                        {`Add ${entityDefinition.entityNameCapitalised}`}
                    </h1>
                </div>
            </div>
            <div className="govuk-grid-row">
                <div className="govuk-grid-column-one-half-from-desktop">
                    <form action={`/api/entity/list/${entityDefinition.entityListName}`} method="POST" onSubmit={handleSubmit}>
                        <input type="hidden" name="_csrf" value={csrfToken} />
                        <Text
                            label={`${entityDefinition.entityNameCapitalised} name`}
                            name="title"
                            type="text"
                            updateState={({ name, value }) => dispatch({ name, value })}
                            value={entity.title}
                        />
                        <Text
                            label={`${entityDefinition.entityNameCapitalised} code`}
                            name="simpleName"
                            type="text"
                            updateState={({ name, value }) => dispatch({ name, value })}
                            value={entity.simpleName}
                        />
                        <Submit />
                    </form>
                </div>
            </div>
        </>
    );
};

const WrappedAddEntity = (entityDefinition: EntityDefinition) => {
    const WrappedAddEntity = ({ history, location, match }: RouteComponentProps) => (
        <ApplicationConsumer>
            {({ csrf }) => (
                <AddEntity
                    csrfToken={csrf} history={history} location={location} match={match}
                    entityDefinition={entityDefinition}/>
            )}
        </ApplicationConsumer>
    );

    return WrappedAddEntity;
};

export default WrappedAddEntity;
