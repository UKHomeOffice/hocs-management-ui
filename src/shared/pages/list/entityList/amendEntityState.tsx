export interface State {
    simpleName: string;
    title: string;
    originalTitle: string;
    uuid: string;
    active: false
}
