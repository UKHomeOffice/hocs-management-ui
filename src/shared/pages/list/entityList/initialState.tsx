import { State } from './state';

export const initialState: State = {
    entities: [],
    entitiesLoaded: false,
    entitiesToDisplay: [],
    showInactive: false,
    inactiveCount: 0
};
