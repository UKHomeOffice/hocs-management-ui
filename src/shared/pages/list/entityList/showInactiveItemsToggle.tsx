import React, { FormEvent, useCallback } from 'react';

interface ShowInactiveItemsToggleProps {
    inactiveCount: number;
    showInactive: boolean;
    onToggle: (flag: boolean) => void;
}

export const ShowInactiveItemsToggle = ({ inactiveCount, showInactive, onToggle }: ShowInactiveItemsToggleProps) => {
    const toggleShowInactive = useCallback((flag: boolean, event: FormEvent) => {
        event.preventDefault();
        onToggle(flag);
    }, [onToggle]);

    return <>
        {inactiveCount > 0 && !showInactive &&
            <p className="govuk-hint">
                {inactiveCount} inactive item{inactiveCount === 1 ? ' is' : 's are'} not being shown.{' '}
                <a href="#" onClick={event => toggleShowInactive(true, event)}>
                    Show inactive items
                </a>.
            </p>
        }
        {inactiveCount > 0 && showInactive &&
            <p className="govuk-hint">
                Showing {inactiveCount} inactive item{inactiveCount === 1 ? '' : 's'}.{' '}
                <a href="#" onClick={event => toggleShowInactive(false, event)}>
                    Hide inactive items
                </a>.
            </p>
        }
    </>;
};
