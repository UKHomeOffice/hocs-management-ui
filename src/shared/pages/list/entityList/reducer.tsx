import { State } from './state';
import { Action } from './actions';

export const reducer = (state: State, action: Action): State => {
    switch (action.type) {
        case 'PopulateEntities':
            return { ...state, entities: action.payload, entitiesLoaded: true };
    }
    return state;
};
