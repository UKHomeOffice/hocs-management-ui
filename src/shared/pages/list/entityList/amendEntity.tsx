import React, { Reducer, useEffect } from 'react';
import { RouteComponentProps } from 'react-router';
import { Link } from 'react-router-dom';
import { object, string } from 'yup';
import Submit from '../../../common/components/forms/submit';
import Text from '../../../common/components/forms/text';
import { ApplicationConsumer } from '../../../contexts/application';
import { updateListItem, getItemDetails } from '../../../services/entityListService';
import { reducer } from './amendEntityReducer';
import ErrorSummary from '../../../common/components/errorSummary';
import {
    GENERAL_ERROR_TITLE,
    VALIDATION_ERROR_TITLE
} from '../../../models/constants';
import useError from '../../../hooks/useError';
import ErrorMessage from '../../../models/errorMessage';
import { validate } from '../../../validation';
import { Action } from './actions';
import { State } from './amendEntityState';

interface MatchParams {
    itemUUID: string;
}
interface AmendEntityProps extends RouteComponentProps<MatchParams> {
    csrfToken?: string;
    entityDefinition: EntityDefinition;
}


const AmendEntity: React.FC<AmendEntityProps> =
        ({ csrfToken, history, match, entityDefinition }) => {
            const initialState: State = {
                uuid: '',
                title: '',
                originalTitle: '',
                simpleName: ''
            };

            const validationSchema = object({
                title: string()
                    .required()
                    .label(`New ${entityDefinition.entityName} name`)
                    .matches(/^[a-zA-Z0-9_,.!? ()&]*$/)
            });

            const [pageError, addFormError, clearErrors, setErrorMessage]
                = useError('', VALIDATION_ERROR_TITLE);
            const [state, dispatch] = React.useReducer<Reducer<State, Action>>(reducer, initialState);

            const { params: { itemUUID } } = match;

            useEffect(() => {
                getItemDetails(itemUUID)
                    .then(item => dispatch({ type: 'SetItemDetails', payload: item }))
                    .catch(() => {
                        setErrorMessage(new ErrorMessage(
                            entityDefinition.messages.LOAD_ENTITIES_ERROR, GENERAL_ERROR_TITLE));
                    });
            }, []);

            const handleSubmit = (event: React.FormEvent) => {
                event.preventDefault();
                clearErrors();
                if (validate(validationSchema, state, addFormError)) {
                    updateListItem(state, entityDefinition.entityListName).then(() => {
                        history.push('/', { successMessage: entityDefinition.messages.AMEND_ENTITY_SUCCESS });
                    }).catch((error) => {
                        setErrorMessage(new ErrorMessage(entityDefinition.messages.AMEND_ENTITY_ERROR_DESCRIPTION, GENERAL_ERROR_TITLE));
                    });
                }
            };

            return (
                <>
                    <div className="govuk-grid-row">
                        <div className="govuk-grid-column-two-thirds-from-desktop">
                            <Link to={entityDefinition.entityRoute} className="govuk-back-link">Back</Link>
                            <ErrorSummary
                                pageError={pageError}
                            />
                            <h1 className="govuk-heading-xl">
                                {`Amend ${entityDefinition.entityName}`}
                            </h1>
                        </div>
                    </div>
                    <div className="govuk-grid-row">
                        <div className="govuk-grid-column-one-half-from-desktop">
                            <h2 className="govuk-heading-l">
                                {`New ${entityDefinition.entityName}: ${state.originalTitle}`}
                            </h2>
                            <form action={`/api/entity/list/update/${entityDefinition.entityListName}`}
                                method="POST" onSubmit={handleSubmit}>
                                <input type="hidden" name="_csrf" value={csrfToken} />
                                <Text
                                    label={`New ${entityDefinition.entityName} name`}
                                    name="title"
                                    type="text"
                                    updateState={({ value }) => dispatch({ type: 'SetTitle', payload: value as string })}
                                    value={state.title}
                                />
                                <Text
                                    label={`${entityDefinition.entityName} code`}
                                    name="simpleName"
                                    type="text"
                                    disabled={true}
                                    updateState={({ value }) => dispatch({ type: 'SetSimpleName', payload: value as string })}
                                    value={state.simpleName}
                                />
                                <Submit />
                            </form>
                        </div>
                    </div>
                </>
            );
        };

const WrappedAmendEntity = (entityDefinition: EntityDefinition) => {
    const WrappedAmendEntity = ( { history, location, match } : RouteComponentProps<MatchParams>) => (
        <ApplicationConsumer>
            {({ csrf }) => (
                <AmendEntity csrfToken={csrf} history={history} location={location} match={match}
                    entityDefinition={entityDefinition}/>
            )}
        </ApplicationConsumer>
    );

    return WrappedAmendEntity;
};

export default WrappedAmendEntity;
