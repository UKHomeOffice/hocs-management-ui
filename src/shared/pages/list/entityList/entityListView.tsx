import React, { Fragment, Reducer, useCallback, useEffect } from 'react';
import { RouteComponentProps } from 'react-router';
import { History } from 'history';
import { getListItems } from '../../../services/entityListService';
import { State } from './state';
import { Action } from './actions';
import { reducer } from './reducer';
import { initialState } from './initialState';
import { GENERAL_ERROR_TITLE } from '../../../models/constants';
import ErrorSummary from '../../../common/components/errorSummary';
import ErrorMessage from '../../../models/errorMessage';
import useError from '../../../hooks/useError';
import { Link } from 'react-router-dom';
import EntityListItem from '../../../models/entityListItem';
import { ShowInactiveItemsToggle } from './showInactiveItemsToggle';

interface MatchParams {
    teamId: string;
}

type EntityListViewProps = RouteComponentProps<MatchParams>;


const EntityListView = (entityDefinition: EntityDefinition) => {
    const onAddClick = (history: History) => {
        history.push(`${entityDefinition.entityRoute}/add`);
    };

    const EntityListView: React.FC<EntityListViewProps> = ( { history } ) => {

        const [pageError, , clearErrors, setErrorMessage] = useError();
        const [state, dispatch] = React.useReducer<Reducer<State, Action>>(reducer, initialState);

        useEffect(() => {
            getListItems(entityDefinition.entityListName)
                .then((entities: EntityListItem[]) => dispatch({ type: 'PopulateEntities', payload: entities }))
                .catch(() => {
                    setErrorMessage(new ErrorMessage(entityDefinition.messages.LOAD_ENTITIES_ERROR,
                        GENERAL_ERROR_TITLE));
                });
        }, []);

        const amendEntity = useCallback((entityUuid: string, event: React.FormEvent) => {
            event.preventDefault();
            clearErrors();
            history.push(`${entityDefinition.entityRoute}/${entityUuid}/amend`);
        }, []);

        const toggleShowInactive = useCallback(
            (showInactive: boolean) => dispatch({ type: 'ToggleShowInactive', payload: showInactive }), [dispatch]
        );

        const DisplayCampaignTable = () => (
            <Fragment>
                {state.entitiesLoaded && (<>
                    <ShowInactiveItemsToggle showInactive={state.showInactive} inactiveCount={state.inactiveCount} onToggle={toggleShowInactive}/>
                    <table className="govuk-table">
                        <thead className="govuk-table__head">
                            <tr className="govuk-table__row">
                                <th className="govuk-table__header" scope="col">{`${entityDefinition.entityNameCapitalised} name`}</th>
                                <th className="govuk-table__header" scope="col">{`${entityDefinition.entityNameCapitalised} code`}</th>
                                {state.showInactive && <th className="govuk-table__header" scope="col">Active?</th>}
                                <th className="govuk-table__header" scope="col">Action</th>
                            </tr>
                        </thead>
                        <tbody className="govuk-table__body">
                            {
                                state.entitiesToDisplay.map((entity) => {
                                    return (
                                        <tr className={`govuk-table__row${entity.active ? '' : ' inactive-entity'}`} key={entity.uuid}>
                                            <td className="govuk-table__cell">{entity.title}</td>
                                            <td className="govuk-table__cell">{entity.simpleName}</td>
                                            {state.showInactive && <td className="govuk-table__cell" scope="col">{entity.active ? 'Yes' : 'No'}</td>}
                                            <td className="govuk-table__cell">
                                                <a href="#" onClick={event => amendEntity(entity.uuid, event)}>Amend</a>
                                            </td>
                                        </tr>
                                    );
                                })
                            }
                        </tbody>
                    </table>
                </>)}
            </Fragment>
        );

        return (
            <div className="govuk-form-group">
                <Link to="/" className="govuk-back-link">Back</Link>
                <ErrorSummary
                    pageError={pageError}
                />
                <div>
                    <h1 className="govuk-heading-xl">{`View and edit ${entityDefinition.entityNamePlural}`}</h1>
                    {
                        state.entitiesLoaded ?
                            <div>
                                <DisplayCampaignTable />
                            </div> :
                            <div>
                                <p className="govuk-body">Loading...</p>
                            </div>
                    }
                    <button type="submit" className="govuk-button govuk-!-margin-right-1 add-team-members-button" data-module="govuk-button" onClick={() => onAddClick(history)}>{`Add new ${entityDefinition.entityName}`}</button>
                </div>
            </div>
        );
    };

    return EntityListView;
};

export default EntityListView;
