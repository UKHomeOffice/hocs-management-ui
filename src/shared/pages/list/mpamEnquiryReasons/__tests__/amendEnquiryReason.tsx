import React from 'react';
import { match, MemoryRouter } from 'react-router-dom';
import { createBrowserHistory, History, Location } from 'history';
import { act, render, RenderResult, wait, fireEvent, waitForElement } from '@testing-library/react';
import {
    AMEND_ENQ_REASON_ERROR_DESCRIPTION,
    DUPLICATE_ENQ_REASON_DESCRIPTION,
    GENERAL_ERROR_TITLE,
    LOAD_ENQ_SUB_ERROR_DESCRIPTION,
    VALIDATION_ERROR_TITLE
} from '../../../../models/constants';
import * as EntityListService from '../../../../services/entityListService';
import * as useError from '../../../../hooks/useError';
import { State } from '../../entityList/amendEntityState';
import AmendEnquiryReason from '../amendEnquiryReason';

let match: match<any>;
let history: History<any>;
let location: Location;
let wrapper: RenderResult;
let mockState: State;

const useReducerSpy = jest.spyOn(React, 'useReducer');
const reducerDispatch = jest.fn();
const useErrorSpy = jest.spyOn(useError, 'default');
const addFormErrorSpy = jest.fn();
const clearErrorsSpy = jest.fn();
const setMessageSpy = jest.fn();

const renderComponent = () => render(
    <MemoryRouter>
        <AmendEnquiryReason history={history} location={location} match={match}></AmendEnquiryReason>
    </MemoryRouter>
);
const getItemDetailsSpy = jest.spyOn(EntityListService, 'getItemDetails');
const updateListItemSpy = jest.spyOn(EntityListService, 'updateListItem');
beforeEach(() => {
    history = createBrowserHistory();
    match = {
        isExact: true,
        params: { itemUUID: '__itemId__' },
        path: '',
        url: ''
    };
    mockState = {
        title: '',
        originalTitle: '',
        simpleName: '',
        uuid: '',
        active: false
    };

    getItemDetailsSpy.mockImplementation(() => Promise.resolve({ simpleName: 'testSimpleName', title: 'testTitle', uuid: 'testUUID', active: false }));
    updateListItemSpy.mockImplementation(() => Promise.resolve());
    useReducerSpy.mockImplementation(() => [mockState, reducerDispatch]);
    useErrorSpy.mockImplementation(() => [{}, addFormErrorSpy, clearErrorsSpy, setMessageSpy]);
    history.push = jest.fn();
    reducerDispatch.mockReset();
    addFormErrorSpy.mockReset();
    clearErrorsSpy.mockReset();
    setMessageSpy.mockReset();
    act(() => {
        wrapper = renderComponent();
    });
});

describe('when the amendEnquiryReason component is mounted', () => {
    it('should render with default props', async () => {
        expect.assertions(2);
        wrapper = renderComponent();
        await wait(() => {
            expect(getItemDetailsSpy).toHaveBeenCalled();
            expect(wrapper.container).toMatchSnapshot();
        });
    });

    it('should initially render blank before item title is returned', async () => {
        getItemDetailsSpy.mockReturnValueOnce(Promise.resolve(
            { simpleName: 'testSimpleName', title: 'testTitle', uuid: 'testUUID', active: false }
        ));
        mockState.originalTitle = '';
        const wrapper: RenderResult = renderComponent();
        expect(wrapper.container.outerHTML).toMatchSnapshot();
    });

    it('should display an error if the call to retrieve item details fails', async () => {
        expect.assertions(1);
        getItemDetailsSpy.mockImplementation(() => Promise.reject('error'));

        wrapper = renderComponent();

        await wait(() => {
            expect(setMessageSpy).toBeCalledWith({ title: GENERAL_ERROR_TITLE, description: LOAD_ENQ_SUB_ERROR_DESCRIPTION });
        });

    });
});

describe('when the submit button is clicked', () => {
    describe('and the data is filled in', () => {

        beforeEach(async () => {
            mockState.title = '__displayName__';
            mockState.simpleName = '__shortCode__';
            const submitButton = await waitForElement(async () => {
                return await wrapper.findByText('Amend');
            });

            fireEvent.click(submitButton);
        });

        describe('and the service call is successful', () => {
            it('should redirect to the home page', async () => {

                getItemDetailsSpy.mockReturnValueOnce(Promise.resolve(
                    { simpleName: 'testSimpleName', title: 'testTitle', uuid: 'testUUID', active: false }
                ));
                await wait(() => {
                    expect(getItemDetailsSpy).toHaveBeenCalled();
                    expect(updateListItemSpy).toHaveBeenCalled();
                    expect(history.push).toHaveBeenCalledWith('/', { successMessage: 'The enquiry reason was amended successfully' });
                });
            });
            it('should call the begin submit action', async () => {

                await wait(() => {
                    expect(clearErrorsSpy).toHaveBeenCalled();
                });
            });
        });
    });
    describe('and the data is not filled in', () => {
        beforeEach(async () => {
            const submitButton = await waitForElement(async () => {
                return await wrapper.findByText('Amend');
            });
            mockState.title = '';
            mockState.simpleName = '';

            fireEvent.click(submitButton);
        });

        it('should call the begin submit action', () => {
            expect(clearErrorsSpy).toHaveBeenCalled();
        });

        it('should set the error state', () => {
            expect(addFormErrorSpy).toHaveBeenNthCalledWith(1, { key: 'title', value: 'The New Enquiry Reason is required' });
        });
    });
});

describe('when the submit button is clicked', () => {
    describe('and the data is filled in', () => {

        beforeEach(async () => {
            updateListItemSpy.mockReset();
            updateListItemSpy.mockImplementation(() => Promise.reject({ response: { status: 500 } }));
            mockState.title = '__displayName__';
            mockState.simpleName = '__shortCode__';
            const submitButton = await waitForElement(async () => {
                return await wrapper.findByText('Amend');
            });

            fireEvent.click(submitButton);
        });

        describe('and the service call fails', () => {
            it('should set the error state', () => {
                expect(setMessageSpy).toHaveBeenCalledWith({ description: AMEND_ENQ_REASON_ERROR_DESCRIPTION, title: GENERAL_ERROR_TITLE });
            });

            it('should call the begin submit action', () => {
                expect(clearErrorsSpy).toHaveBeenCalled();
            });
        });
    });

    describe('and the data already exists', () => {
        beforeEach(async () => {
            updateListItemSpy.mockReset();
            updateListItemSpy.mockImplementationOnce(() => Promise.reject({ response: { status: 409 } }));
            mockState.title = '__displayName__';
            mockState.simpleName = '__shortCode__';
            const submitButton = await waitForElement(async () => {
                return await wrapper.findByText('Amend');
            });

            fireEvent.click(submitButton);
        });

        it('should set the error state', () => {
            expect(setMessageSpy).toHaveBeenCalledWith({ description: DUPLICATE_ENQ_REASON_DESCRIPTION, title: VALIDATION_ERROR_TITLE });
        });

        it('should call the begin submit action', () => {
            expect(clearErrorsSpy).toHaveBeenCalled();
        });
    });
});
