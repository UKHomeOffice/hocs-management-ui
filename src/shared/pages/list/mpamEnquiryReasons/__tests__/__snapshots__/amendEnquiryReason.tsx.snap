// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when the amendEnquiryReason component is mounted should initially render blank before item title is returned 1`] = `"<div><div class=\\"govuk-grid-row\\"><div class=\\"govuk-grid-column-two-thirds-from-desktop\\"><a class=\\"govuk-back-link\\" href=\\"/enquiry-subject/undefined\\">Back</a><h1 class=\\"govuk-heading-xl\\">Amend Enquiry Reason</h1></div></div><div class=\\"govuk-grid-row\\"><div class=\\"govuk-grid-column-one-half-from-desktop\\"><h2 class=\\"govuk-heading-l\\">Enquiry Reason: </h2><form action=\\"/api/entity/list/update/undefined\\" method=\\"POST\\"><input type=\\"hidden\\" name=\\"_csrf\\" value=\\"\\"><div class=\\"govuk-form-group\\"><label for=\\"title\\" id=\\"title-label\\" class=\\"govuk-label govuk-label--s\\">New Enquiry Reason</label><input class=\\"govuk-input\\" id=\\"title\\" type=\\"text\\" name=\\"title\\" value=\\"\\"></div><input class=\\"govuk-button\\" type=\\"submit\\" value=\\"Amend\\"></form></div></div></div>"`;

exports[`when the amendEnquiryReason component is mounted should render with default props 1`] = `
<div>
  <div
    class="govuk-grid-row"
  >
    <div
      class="govuk-grid-column-two-thirds-from-desktop"
    >
      <a
        class="govuk-back-link"
        href="/enquiry-subject/undefined"
      >
        Back
      </a>
      <h1
        class="govuk-heading-xl"
      >
        Amend Enquiry Reason
      </h1>
    </div>
  </div>
  <div
    class="govuk-grid-row"
  >
    <div
      class="govuk-grid-column-one-half-from-desktop"
    >
      <h2
        class="govuk-heading-l"
      >
        Enquiry Reason: 
      </h2>
      <form
        action="/api/entity/list/update/undefined"
        method="POST"
      >
        <input
          name="_csrf"
          type="hidden"
          value=""
        />
        <div
          class="govuk-form-group"
        >
          <label
            class="govuk-label govuk-label--s"
            for="title"
            id="title-label"
          >
            New Enquiry Reason
          </label>
          <input
            class="govuk-input"
            id="title"
            name="title"
            type="text"
            value=""
          />
        </div>
        <input
          class="govuk-button"
          type="submit"
          value="Amend"
        />
      </form>
    </div>
  </div>
</div>
`;
